api/src/main/java/org/apache/iceberg/actions/compaction/BinPack.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions.compaction;
+
+public interface BinPack {
```

Comment:
Would it make sense to add this one once we have a bit more clarity on the implementation? Right now, it is only used in the default compaction strategy constant. We can probably add it later and focus on the common API for now.

Just asking.

Reply:
Why not an enum for the strategies?

Reply:
This should probably also be private. There's no need to expose this, I think.

Reply:
Yeah sorry this was just a placeholder, I think an Enum would probably be fine. We'll have to determine the actual "BinPack" at the action level since we'll need a different class depending on the executing framework.

Reply:
I think my initial intention was to put Bin-pack specific options in here, but I think we can just keep them in the base class.

Reply:
I'd support an idea of an enum or something in `core` and not exposing it for now.